### PR TITLE
Wiremod port

### PR DESCRIPTION
**Pull request by Archanial**
## About The Pull Request

PRs ported so far:
https://github.com/tgstation/tgstation/pull/59348
https://github.com/tgstation/tgstation/pull/59459
https://github.com/tgstation/tgstation/pull/59584
https://github.com/tgstation/tgstation/pull/59592
https://github.com/tgstation/tgstation/pull/59613
https://github.com/tgstation/tgstation/pull/59345	
https://github.com/tgstation/tgstation/pull/58419 (only few parts)
https://github.com/tgstation/tgstation/pull/59728
https://github.com/tgstation/tgstation/pull/59724
https://github.com/tgstation/tgstation/pull/59743
https://github.com/tgstation/tgstation/pull/59850
https://github.com/tgstation/tgstation/pull/59635
https://github.com/tgstation/tgstation/pull/59866

map changes included 

This is dependent on #4623 and #4678

## Changelog
:cl: Watermelon914, Mothblocks
add: Wiremod, circuit replacement.
/:cl:

### CODE REVIEW

**Comment by ike709** [comment]
What about https://github.com/tgstation/tgstation/pull/59635?

Also https://github.com/tgstation/tgstation/pull/59866 when it's merged

**Comment by Archanial** [comment]
First one is included second one will be added to this pr or ported separately when it gets merged on tg.

File: code/game/objects/items/devices/multitool.dm [hunk 1]
```diff
@@ -30,15 +30,22 @@
 	usesound = 'sound/weapons/empty.ogg'
 	var/mode = 0
 
+/obj/item/multitool/Initialize()
+	RegisterSignal(src, COMSIG_PARENT_EXAMINE, .proc/on_examine)
+	return ..()
+
+/obj/item/multitoo/Destroy()
```
Comment (ike709) [inline code/game/objects/items/devices/multitool.dm]: Dude
```suggestion
/obj/item/multitool/Destroy()
```

**Comment by MNarath1** [comment]
There is one thing that i found during testmerge the input for if the component is active will always display on even trough its logically set to 0 so off it should display on on 1 and off on 0 correctly.

**Comment by BrumalLake** [comment]
airlock shell doesnt seem to work
cant put in circuit, try screwdriving and it doesnt show a do after bar or anything it just says you begin screwing and makes a noise

**Comment by Archanial** [comment]
> 
> 
> airlock shell doesnt seem to work
> cant put in circuit, try screwdriving and it doesnt show a do after bar or anything it just says you begin screwing and makes a noise

airlock shells works. The issue you are talking about is progress bars not appearing. The airlock will finish after 10 seconds but lack of feedback is bad.
If you want to insert circuit into airlock just attack it with circuit on harm.

**Review by PowerfulBacon — APPROVED** [review]
corg map changes fine lol

**Pull request merged**